src/components/MagicLinkSuccessContent.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react';
+import { Trans } from 'react-i18next';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import { RecaptchaAction } from '@graasp/sdk';
+import { namespaces } from '@graasp/translations';
+import { Button } from '@graasp/ui';
+
+import MailOutlineIcon from '@mui/icons-material/MailOutline';
+import { Container, Stack, Typography } from '@mui/material';
+
+import { useAuthTranslation } from '../config/i18n';
+import { mutations } from '../config/queryClient';
+import {
+  BACK_BUTTON_ID,
+  RESEND_EMAIL_BUTTON_ID,
+  SUCCESS_CONTENT_ID,
+} from '../config/selectors';
+import { useRecaptcha } from '../context/RecaptchaContext';
+import { useRedirection } from '../hooks/searchParams';
+import { AUTH } from '../langs/constants';
+import FullscreenContainer from './FullscreenContainer';
+
+const MagicLinkSuccessContent = () => {
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+  const { t } = useAuthTranslation();
+  const { executeCaptcha } = useRecaptcha();
+  const [isEmailSent, setIsEmailSent] = useState(false);
+
+  // used for resend email
+  const redirect = useRedirection();
+  const { mutate: signIn } = mutations.useSignIn();
+
+  const email = searchParams.get('email');
+
+  // used for resend email
+  const handleResendEmail = async () => {
+    const lowercaseEmail = email.toLowerCase();
+    const token = await executeCaptcha(RecaptchaAction.SignIn);
+    signIn({
+      email: lowercaseEmail,
+      captcha: token,
+      url: redirect.url,
+    });
+  };
+
+  const onClickResendEmail = () => {
+    setIsEmailSent(true);
+    handleResendEmail();
+  };
+
+  const handleBackButtonClick = () => {
+    navigate(-1);
```

Comment:
is this means navigating back ? Maybe using a const like `const NAVIGATE_BACK = -1` could help increasing the understanding here.

```suggestion
    navigate(NAVIGATE_BACK);
```
```suggestion
    navigate(-1);
```